pkg/engine/query/source_test.go

```diff
@@ -0,0 +1,317 @@
+			},
+			contains: "generic.k8s",
+			wantErr:  false,
+		},
+		{
+			name: "get_generic_query_unknown",
+			fields: fields{
+				Source: "./assets/queries/template",
+			},
+			args: args{
+				platform: "unknown",
+			},
+			contains: "generic.common",
+			wantErr:  false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			s := &FilesystemSource{
+				Source: tt.fields.Source,
+			}
+			got, err := s.GetGenericQuery(tt.args.platform)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("FilesystemSource.GetGenericQuery() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !strings.Contains(got, tt.contains) {
+				t.Errorf("FilesystemSource.GetGenericQuery() = %v, doesn't contains %v", got, tt.contains)
+			}
+		})
+	}
+}
+
+func TestFilesystemSource_GetQueries(t *testing.T) {
+	changeCurrentDir(t)
+
+	type fields struct {
+		Source string
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		want    []model.QueryMetadata
+		wantErr bool
+	}{
+		{
+			name: "get_queries_1",
+			fields: fields{
+				Source: "./assets/queries/template",
+			},
+			want: []model.QueryMetadata{
+				{
+					Query: "template",
+					Content: `package Cx
+
+CxPolicy [ result ] {
+  resource := input.document[i].resource
+  resource == "<VALUE>"
+
+	result := {
+                "documentId": 		input.document[i].id,
+                "searchKey": 	    sprintf("%s", [resource]),
+                "issueType":		"IncorrectValue",  #"MissingAttribute" / "RedundantAttribute"
+                "keyExpectedValue": "<RESOURCE>",
+                "keyActualValue": 	resource
+              }
+}`,
+					Metadata: map[string]interface{}{
+						"category":        nil,
+						"descriptionText": "<TEXT>",
+						"descriptionUrl":  "#",
+						"id":              "<ID>",
+						"queryName":       "<QUERY_NAME>",
+						"severity":        "HIGH",
+					},
+					Platform: "unknown",
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "get_queries_error",
+			fields: fields{
+				Source: "../no-path",
+			},
+			want:    nil,
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			s := &FilesystemSource{
+				Source: tt.fields.Source,
+			}
+			got, err := s.GetQueries()
+			if (err != nil) != tt.wantErr {
+				t.Errorf("FilesystemSource.GetQueries() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			require.Equal(t, tt.want, got)
+		})
+	}
+}
+
+func changeCurrentDir(t *testing.T) {
+	for currentDir, err := os.Getwd(); getCurrentDirName(currentDir) != "kics"; currentDir, err = os.Getwd() {
+		if err == nil {
+			if err := os.Chdir(".."); err != nil {
+				fmt.Printf("change path error = %v", err)
+				t.Fatal()
+			}
+		} else {
+			t.Fatal()
+		}
+	}
+}
+
+func getCurrentDirName(path string) string {
+	dirs := strings.Split(path, string(os.PathSeparator))
+	if dirs[len(dirs)-1] == "" && len(dirs) > 1 {
+		return dirs[len(dirs)-2]
+	}
+	return dirs[len(dirs)-1]
+}
```

Comment:
functions being used in `pkg/engine/inspector_test.go`. Make these functions inside a common helper file

Reply:
resolved

Reply:
Moved to test/helpers.go